mesteriApplication/src/main/java/com/intern/alexx/repository/impl/MesterRepositoryImp.java

```diff
@@ -15,234 +14,248 @@
+		int totalElements = 0;
 		try {
-			conn = dataSource.getConnection();
-			PreparedStatement ps = conn.prepareStatement(sql1);
-			PreparedStatement ps2 = conn.prepareStatement(sql2);
-		
-			int nrParm=1;
-				if (searchCriteria.getFirstName() != null) {
-					 ps.setString(nrParm, searchCriteria.getFirstName());
-					 ps2.setString(nrParm, searchCriteria.getFirstName());
-					 nrParm++;
-					 }
-				if (searchCriteria.getLastName() != null) {
-					 ps.setString(nrParm, searchCriteria.getLastName());
-					 ps2.setString(nrParm, searchCriteria.getLastName());
-					 nrParm++;
-					 }
-				if (searchCriteria.getLocation() != null) {
-					 ps.setString(nrParm, searchCriteria.getLocation());
-					 ps2.setString(nrParm, searchCriteria.getLocation());
-					 nrParm++;
-					 }
-				if (searchCriteria.getSpecialityName() != null) {
-					 ps.setString(nrParm, searchCriteria.getSpecialityName());
-					 ps2.setString(nrParm, searchCriteria.getSpecialityName());
-					 nrParm++;
-					 }
-				if (searchCriteria.getEmail() != null) {
-					 ps.setString(nrParm, searchCriteria.getEmail());
-					 ps2.setString(nrParm, searchCriteria.getEmail());
-					 nrParm++;
-					 }
-				if (searchCriteria.getPhoneNumber() != null) {
-					 ps.setString(nrParm, searchCriteria.getPhoneNumber());
-					 ps2.setString(nrParm, searchCriteria.getPhoneNumber());
-					 nrParm++;
-					 }
-				if (searchCriteria.getRating() != null) {
-					 ps.setInt(nrParm, searchCriteria.getRating());
-					 ps2.setInt(nrParm, searchCriteria.getRating());
-					 nrParm++;
-					  }
-				if (searchCriteria.getPrice() != null) {
-					 ps.setString(nrParm, searchCriteria.getPrice());
-					 ps2.setString(nrParm, searchCriteria.getPrice());
-					 nrParm++;
-					 
-				}
-				
-				if(searchCriteria.getPageNumber() != null) {
-				page.setPageNumber(searchCriteria.getPageNumber());
-				}
-				if(searchCriteria.getPageSize() != null) {
-				page.setPageSize(searchCriteria.getPageSize());
-				}
-			ResultSet resultSet = ps.executeQuery();
-		
-			page.setTotalRezults(resultSet.getInt(1));
-			ps.close();
-			
-			ResultSet resultSet2 = ps2.executeQuery();
-			if (resultSet2.next()) {
-				mester = new Mester();
-				mester.setId(resultSet2.getInt("id"));
-				mester.setFirstName(resultSet2.getString("first_name"));
-				mester.setLastName(resultSet2.getString("last_name"));
-				mester.setDescription(resultSet2.getString("description"));
-				mester.setLocation(resultSet2.getString("location"));
-				page.getContentPage().add(mester);
+
+			ps = connectionUtil.prepareConnection(conn, sql);
+			verifyParam(ps, searchCriteria);
+			resultSet = ps.executeQuery();
+
+			if (resultSet.next()) {
+				totalElements = resultSet.getInt("total");
 			}
-			page.getContentPage();
-			ps2.close();
-		} catch (
 
-		SQLException e)
+		} catch (SQLException e) {
+			throw new RepositoryException(e);
+		} finally {
+			try {
```

Comment:
Muta try/catch in closeConnection() ...   numele nu e bun ..  inchizi si statement si result set. Fa atunci o metoda in care sa inchizi in Closeable... si o apelezi de 3 ori